### DISCUSSION TITLE
Can it be used in Unity's WebGL?

### DISCUSSION THREAD

**Discussion by a13782425** [timcassell/ProtoPromise#446] [Q&A] [answered]
Can I use it in Unity's WebGL?

**Comment by timcassell** [comment, marked as answer]
Yes, Unity's WebGL is supported!

The only thing that doesn't work is background threads. You can assign a `SynchronizationContext` to `Promise.Config.BackgroundContext` if you need to support multiple platforms and you're using background threads on other platforms.

**Reply by a13782425** [reply]
When I need to support background threads,, gave `Promise.Config.BackgroundContext` assigned a background `SynchronizationContext`?

**Reply by timcassell** [reply]
Unity WebGL doesn't support background threads. You would need to assign a context that executes on the main thread. You can implement your own context (easy one is to just invoke the callback synchronously in `SynchronizationContext.Post`). Or you could even just assign the same as the `ForegroundContext` if you would rather not implement your own. `Promise.Config.BackgroundContext = Promise.Config.ForegroundContext`

**Reply by a13782425** [reply]
Thank you!
